pkg/registry/servicecatalog/binding/tpr_storage_interface.go

```diff
@@ -0,0 +1,276 @@
+)
+
+type tprStorageInterface struct {
+	cl        *dynamic.Client
+	decodeKey func(string) (string, string, error)
+}
+
+func newTPRStorageInterface(
+	cl *dynamic.Client,
+	decodeKeyFn func(string) (string, string, error),
+) (storage.Interface, factory.DestroyFunc) {
+	return &tprStorageInterface{cl: cl}, factory.DestroyFunc(func() {})
+}
+
+// Versioned returns the versioned associated with this interface
+func (t *tprStorageInterface) Versioner() storage.Versioner {
+	return &tprStorageVersioner{cl: t.cl}
+}
+
+// Create adds a new object at a key unless it already exists. 'ttl' is time-to-live
+// in seconds (0 means forever). If no error is returned and out is not nil, out will be
+// set to the read value from database.
+func (t *tprStorageInterface) Create(
+	ctx context.Context,
+	key string,
+	obj,
+	out runtime.Object,
+	ttl uint64,
+) error {
+	objBinding, ok := obj.(*servicecatalog.Binding)
+	if !ok {
+		return errObjNotABinding
+	}
+	unstruc, err := tpr.ToUnstructured(objBinding)
+	if err != nil {
+		return err
+	}
+	cl := tpr.GetResourceClient(t.cl, tpr.ServiceBindingKind, objBinding.Namespace)
+	createdUnstruc, err := cl.Create(unstruc)
+	if err != nil {
+		return err
+	}
+	if err := tpr.FromUnstructured(createdUnstruc, obj); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Delete removes the specified key and returns the value that existed at that spot.
+// If key didn't exist, it will return NotFound storage error.
+//
+// In this implementation, Delete will not write the deleted object back to out
+func (t *tprStorageInterface) Delete(
+	ctx context.Context,
+	key string,
+	out runtime.Object,
+	preconditions *storage.Preconditions,
+) error {
+	ns, name, err := t.decodeKey(key)
+	if err != nil {
+		return err
+	}
+	cl := tpr.GetResourceClient(t.cl, tpr.ServiceBindingKind, ns)
+	if err := cl.Delete(name, nil); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Watch begins watching the specified key. Events are decoded into API objects,
+// and any items selected by 'p' are sent down to returned watch.Interface.
+// resourceVersion may be used to specify what version to begin watching,
+// which should be the current resourceVersion, and no longer rv+1
+// (e.g. reconnecting without missing any updates).
+// If resource version is "0", this interface will get current object at given key
+// and send it in an "ADDED" event, before watch starts.
+func (t *tprStorageInterface) Watch(
+	ctx context.Context,
+	key string,
+	resourceVersion string,
+	p storage.SelectionPredicate,
+) (watch.Interface, error) {
+	ns, name, err := t.decodeKey(key)
+	if err != nil {
+		return nil, err
+	}
+	cl := tpr.GetResourceClient(t.cl, tpr.ServiceBindingKind, ns)
+	binding := servicecatalog.Binding{
+		TypeMeta: metav1.TypeMeta{
+			Kind: tpr.ServiceBindingKind.String(),
+		},
+		ObjectMeta: api.ObjectMeta{
+			Namespace: ns,
+			Name:      name,
+		},
+	}
+	return cl.Watch(&binding)
```

Comment:
TODO: add filter here to convert `*runtime.Unstructured` types into `ServiceBinding`s